xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-security/src/test/it/org/xwiki/test/security/ExtensionSecurityTest.java

```diff
@@ -69,6 +82,59 @@ public void checkForExtensionSecurityScanResult()
         });
     }
 
+    private static void installExtensionSecurityUI() throws Exception
+    {
+        TestUtils testUtils = new TestUtils();
+        String xwikiRESTURL = String.format("%srest", getUtil().getBaseURL());
+
+        // Install the extensions
+        try {
+            ExtensionId extensionId =
+                new ExtensionId("org.xwiki.platform:xwiki-platform-extension-security-ui", testUtils.getVersion());
+            InstallRequest installRequest =
+                createInstallRequest(extensionId, List.of("wiki:" + testUtils.getCurrentWiki()),
+                    ADMIN_CREDENTIALS.getUserName());
+
+            EmbeddableComponentManager componentManager = new EmbeddableComponentManager();
+            componentManager.initialize(ExtensionSecurityTest.class.getClassLoader());
+
+            JobExecutor jobExecutor = new JobExecutor();
+            JobRequest request =
+                componentManager.<ModelFactory>getInstance(ModelFactory.class).toRestJobRequest(installRequest);
+            jobExecutor.execute(InstallJob.JOBTYPE, request, xwikiRESTURL, ADMIN_CREDENTIALS);
+        } catch (Exception e) {
+            throw new Exception(String.format("Failed to install Extension(s) into XWiki at [%s]", xwikiRESTURL), e);
+        }
+    }
+
+    private static InstallRequest createInstallRequest(ExtensionId extensionId, List<String> namespaces,
+        String installUserReference)
+    {
+        InstallRequest installRequest = new InstallRequest();
+
+        // Set a job id to save the job result
+        installRequest.setId("extension", "provision", UUID.randomUUID().toString());
+
+        installRequest.setInteractive(false);
+        installRequest.setFailOnExist(true);
+
+        // Set the extension list to install
+
+        installRequest.addExtension(extensionId);
+
+        // Set the namespaces into which to install the extensions
+        for (String namespace : namespaces) {
+            installRequest.addNamespace(namespace);
+        }
+
+        // Set any user for installing pages (if defined)
+        if (installUserReference != null) {
+            installRequest.setProperty("user.reference",
+                new DocumentReference("xwiki", "XWiki", ADMIN_CREDENTIALS.getUserName()));
+        }
+        return installRequest;
+    }
```

Comment:
For now this is badly duplicated from `xwiki-platform-test-docker` while still requiring to depend on this. I'm not sure where to place a factorized version of this.

Reply:
Sounds like a job for `xwiki-platform-extension-test-utils`.

Reply:
hum, wouldn't that just more the dependency to `xwiki-platform-test-docker` in `xwiki-platform-extension-test-utils`, (or make `xwiki-platform-test-docker` depends on `xwiki-platform-extension-test-utils` which is not great)?

Reply:
> which is not great

Why ?

The only point of this module is to provide tools to manipulate extensions during tests, which is exactly the need of both your module and `xwiki-platform-test-docker`. Having a module that does not need docker at all depends on `xwiki-platform-test-docker` is the one that is not great IMO.

Reply:
Because of `JobExecutor`. I've moved the common code in `xwiki-platform-test-rest` which is now used directly by `xwiki-platform-distribution-flavor-test-security` and internally by `xwiki-platform-test-docker`